frontend/src/test/scala/bloop/dap/DebugProtocolSpec.scala

```diff
@@ -0,0 +1,197 @@
+        |    println("Non-blocking Hello!")
+        |  }
+        |}
+     """.stripMargin
+
+    TestUtil.withinWorkspace { workspace =>
+      val main =
+        """|/main/scala/Main.scala
+           |object Main {
+           |  def main(args: Array[String]): Unit = {
+           |    println("Blocking Hello!")
+           |    synchronized(wait())
+           |  }
+           |}
+           |""".stripMargin
+
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val project = TestProject(workspace, "p", List(main))
+
+      loadBspState(workspace, List(project), logger) { state =>
+        // start debug session and the immediately disconnect from it
+        val blockingSessionOutput = state.withDebugSession(project, mainClassParams("Main")) {
+          client =>
+            for {
+              _ <- client.initialize()
+              _ <- client.launch()
+              _ <- client.configurationDone()
+              output <- client.firstOutput
+              _ <- client.disconnect()
+            } yield output
+        }
+
+        assertNoDiff(blockingSessionOutput, "Blocking Hello!")
+
+        // fix the main class
+        val sources = state.toTestState.getProjectFor(project).sources
+        val mainFile = sources.head.resolve("Main.scala")
+        Files.write(mainFile.underlying, correctMain.getBytes(StandardCharsets.UTF_8))
+
+        // start the next debug session
+        val output = state.withDebugSession(project, mainClassParams("Main")) { client =>
+          for {
+            _ <- client.initialize()
+            _ <- client.launch()
+            _ <- client.configurationDone()
+            _ <- client.exited
+            _ <- client.terminated
+            _ <- client.disconnect()
+            output <- client.allOutput
+          } yield output
+        }
+
+        assertNoDiff(output, "Non-blocking Hello!")
+      }
+    }
+  }
+
+  test("starts a test suite") {
+    TestUtil.withinWorkspace { workspace =>
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      loadBspBuildFromResources("cross-test-build-0.6", workspace, logger) { build =>
+        val project = build.projectFor("test-project-test")
+        val testFilters = List(
+          "hello.JUnitTest",
+          "hello.ScalaTestTest",
+          "hello.ScalaCheckTest",
+          "hello.WritingTest",
+          "hello.Specs2Test",
+          "hello.EternalUTest"
+        )
+
+        val output = build.state.withDebugSession(project, testSuiteParams(testFilters)) { client =>
+          for {
+            _ <- client.initialize()
+            _ <- client.launch()
+            _ <- client.configurationDone()
+            _ <- client.exited
+            _ <- client.terminated
+            _ <- client.disconnect()
+            output <- client.allOutput
+          } yield output
+        }
+
+        val lines = output.lines.toSeq
+        assert(lines.size == 2)
+        assert(lines(0).contains("Running Tests"))
```

Comment:
1. because of a bug ;)
2. Now, we use them to verify the output
3. those are exactly the same kind of filters as expected by the `buildTarget/test` endpoints
4. I am not sure, I could not reproduce. Can you verify it still happens with the newest version?